src/dgrid/dgridConstructorFactory.ts

```diff
@@ -0,0 +1,122 @@
+import { DgridInnerWrapperProperties } from './DgridInnerWrapperProperties';
+import * as declare from 'dojo/_base/declare';
+import * as Grid from 'dgrid/Grid';
+import * as OnDemandGrid from 'dgrid/OnDemandGrid';
+import * as Keyboard from 'dgrid/Keyboard';
+import * as Pagination from 'dgrid/extensions/Pagination';
+import * as Selection from 'dgrid/Selection';
+import * as CellSelection from 'dgrid/CellSelection';
+import * as Selector from 'dgrid/Selector';
+import * as ColumnHider from 'dgrid/extensions/ColumnHider';
+import * as ColumnReorder from 'dgrid/extensions/ColumnReorder';
+import * as ColumnResizer from 'dgrid/extensions/ColumnResizer';
+import * as CompoundColumns from 'dgrid/extensions/CompoundColumns';
+import * as ColumnSet from 'dgrid/ColumnSet';
+import * as Tree from 'dgrid/Tree';
+import { DgridWrapperFeatures, SelectionType } from './DgridWrapperProperties';
+import * as MemoryStore from 'dstore/Memory';
+import * as TreeStore from 'dstore/Tree';
+import * as Trackable from 'dstore/Trackable';
+
+export function buildConstructor(properties: DgridInnerWrapperProperties, emitGridState: () => void): any {
+	const {
+		pagination,
+		keyboard,
+		selection,
+		selector,
+		tree,
+		columnHider,
+		columnReorder,
+		columnResizer,
+		compoundColumns,
+		columnSet
+	} = properties.features || {
```

Comment:
This will only default if `features` is undefined but not if there a user adds just one of the features... all the rest will be undefined (that might be okay, but I am not sure.)